src/pveclib/vec_int128_ppc.h

```diff
@@ -4371,6 +4373,63 @@ vec_cmul100ecuq (vui128_t *cout, vui128_t a, vui128_t cin)
   return ((vui128_t) t);
 }
 
+/** \brief Vector Multiply-Sum and Write Carry-out Unsigned Doubleword.
+ *
+ *  Compute the even and odd 128-bit products of doubleword 64-bit
+ *  element values from a, b.
+ *  Then compute the carry-out of the low order 128-bits of the sum of
+ *  (a<SUB>even</SUB> * b<SUB>even</SUB>) +
+ *  (a<SUB>odd</SUB> * b<SUB>odd</SUB>) + c.
+ *  Only the high order 128 bits of the Multiply-Sum are returned and
+ *  the low order 128-bits of the sum is ignored/lost.
+ *  ResultS are in the range 0-2.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   | 30-32 | 1/cycle  |
+ *  |power9   | 5-7   | 2/cycle  |
+ *
+ *  @param a 128-bit __vector unsigned long int.
+ *  @param b 128-bit __vector unsigned long int.
+ *  @param c 128-bit __vector unsigned __int128.
+ *  @return The carry-out of the __vector unsigned multiply Sum.
+ */
+static inline vui128_t
+vec_msumcud (vui64_t a, vui64_t b, vui128_t c)
+{
+  vui128_t res;
+#if defined (_ARCH_PWR10) && (__GNUC__ >= 10)
```

Comment:
I guess doing something better here would require `autoconfig` magic, which maybe isn't worth the effort. ok.

Reply:
So there is a configure test for PVECLIB_DISABLE_POWER10

This is set when POWER10 support is NOT THERE at any (compiler, assembler, etc) level. Used mostly for disabling compiler or unit tests.

For platform tuning of operations, I use this form because there are often additional constraints on the toolchain (like it will support some P10 instructions, but not this one).

I will leave this for now.